code/obj/machinery/plantpot.dm

```diff
@@ -741,30 +717,34 @@
 			pingsignal.transmission_method = TRANSMISSION_RADIO
 
 			var/datum/radio_frequency/frequency = radio_controller.return_frequency("[report_freq]")
-			if (!frequency) return
+			if(!frequency) return
 			SPAWN_DBG(0.5 SECONDS) //Send a reply for those curious jerks
 				frequency.post_signal(src, pingsignal)
 
 		return //Just toss out the rest of the signal then I guess
 
 	// Procs specific to the plantpot start here.
 
-	proc/update_water()
+	proc/update_water_icon()
+		var/datum/color/average
 		src.water_sprite = image('icons/obj/hydroponics/hydroponics.dmi',"wat-[src.total_volume]")
 		src.water_display = image('icons/obj/hydroponics/hydroponics.dmi',"ind-wat-[src.water_level]")
 
-		if (src.reagents.total_volume)
-			var/datum/color/average = src.reagents.get_average_color()
-			src.water_sprite.color = average.to_rgba()
+		if(src.reagents.total_volume)
+			average = src.reagents.get_average_color()
+			if(average.to_rgba() != src.water_sprite.color)
+				src.water_sprite.color = average.to_rgba()
+			else
+				average = null
 
-		UpdateOverlays(src.water_display, "water")
-		UpdateOverlays(src.water_sprite, "water_fluid")
+		if(average || !current) //only update water level if there is no plant in the tray or reagent contents change color (cheap fix to a layer stacking issue)
+			UpdateOverlays(src.water_sprite, "water_fluid")
+		UpdateOverlays(src.water_display, "water_display")
 
-	proc/update_icon()
-		src.water_sprite = image('icons/obj/hydroponics/hydroponics.dmi',"wat-[src.total_volume]")
+	proc/update_icon() //plant icon stuffs
 		src.water_display = image('icons/obj/hydroponics/hydroponics.dmi',"ind-wat-[src.water_level]")
-		UpdateOverlays(water_display, "water")
-		if (!src.current)
+		UpdateOverlays(water_display, "water_display")
```

Comment:
Just call `update_water_icon`? With an appropriate check before or inside.

Reply:
Water display is separate from water level now. That's why this is handled here :P